test/utils.cpp

```diff
@@ -119,6 +119,11 @@ TEST_CASE("Tokenizers behave correctly", "[utils]") {
 	}
 }
 
+TEST_CASE("utils::consolidate_whitespace bahaves correctly", "[utils]") {
+	std::string result = utils::consolidate_whitespace("    Lorem \t\tIpsum ");
```

Comment:
This doesn't demonstrate how `consolidate_whitespace` handles:

1. trailing whitespace;
2. `\r` and `\n`;
3. inputs where there's nothing to consolidate;
4. being passed a third argument.

Can you expand the tests to cover these cases?